network/groups.go

```diff
@@ -83,72 +84,68 @@ func NewAdversaryGroups() (groups AdversaryGroups) {
 			delay = config.AdversaryDelays[i]
 		}
 
-		groups = append(groups, &AdversaryGroup{
-			NodeIDs:       make([]int, 0),
-			TargetMana:    targetMana,
-			Delay:         time.Millisecond * time.Duration(delay),
-			AdversaryType: ToAdversaryType(configAdvType),
-		})
+		if len(config.AdversaryNodeCounts) > 0 {
```

Comment:
what happens when number of adversary node counts and adversary types is different? shoudn't it panic?

Reply:
I have it covered during parsing this way:
```
		if len(config.AdversaryNodeCounts) != 0 && len(config.AdversaryNodeCounts) != len(config.AdversaryTypes) {
		config.AdversaryNodeCounts = []int{}
	}
```

Reply:
So if the length is not the same for all the parameters, it falls back to using 2 random color issuers?

Reply:
in this case the default values (node count = 1) for AdversaryNodeCounts will be used with warning